### DISCUSSION TITLE
Max number of rows (records)

### DISCUSSION THREAD

**Discussion by IvoSkolek** [marianfoo/ui5-cc-excelUpload#457] [Ideas]
Hi I started using spreadsheet importer and works great, thank you very much! Would be nice if there is property in configuration to set max. number of rows that can be in the file and if exceeded, can't progress further from the dialog. In my application I want to limit max. number of records to 500 in 1 file because of performance reasons.

**Comment by marianfoo** [comment]
Hi Ivo,
thanks for creating a GitHub account just for this suggestion :)
I already have this idea on the roadmap:
https://github.com/users/marianfoo/projects/2

When somebody suggest it, i usually also implement it, also because this should be rather simple.
But i need to think about it.

But just as a suggestion how you could build this on your on:
You can use the Event [`checkBeforeRead `](https://docs.spreadsheet-importer.com/pages/Events/#check-data-before-upload-to-app) to implement your own check, including row size limit.
It could look like this:
```js
// event to check before uploaded to app
this.spreadsheetUpload.attachCheckBeforeRead(function (oEvent) {
// example
const sheetData = oEvent.getParameter("sheetData");
let errorArray = [];
// if sheetdata greater then 500 rows,throw error
if (sheetData.length > 500) {
	const error = {
		title: "Too many rows",
		description: "Please upload less than 500 rows",
		ui5type: "Error"
	};
	errorArray.push(error);
	oEvent.getSource().addArrayToMessages(errorArray);
}
}, this);
```

![image](https://github.com/marianfoo/ui5-cc-spreadsheetimporter/assets/13335743/6e79738e-1e28-468a-8082-a52cd685816b)
